fixed osx input event

toger5 opened the pull request:
godot compiles on osx and user into works.

I have to questions about the code. you can find them in comments searching for toger.
(I now this is a really inelegant way of asking them, ill do a rebase afterwards to remove them)

the function get_osx_key_modifier_state is static. Is that fine?

Handles #8849 for OSX.

toger5 committed:
fixed osx input event

akien-mga commented on platform/osx/os_osx.mm:
```diff
@@ -63,15 +63,23 @@ static NSRect convertRectToBacking(NSRect contentRect) {
-	mod.control = (flags & NSControlKeyMask);
-	mod.alt = (flags & NSAlternateKeyMask);
-	mod.meta = (flags & NSCommandKeyMask);
+//static InputModifierState translateFlags(NSUInteger flags) {
```
Please remove the obsoleted code too.

toger5 commented:
I removed the old code, but there are still the comments where I have questions.
I will do another force pushed where I removed them. But I would be super interested in answers for those questions.

akien-mga commented on platform/osx/os_osx.mm:
```diff
@@ -1516,7 +1522,7 @@ static void keyboardLayoutChanged(CFNotificationCenterRef center, void *observer
 
 void OS_OSX::push_input(const Ref<InputEvent> &p_event) {
-
+	//toger why is ther another function alternative would be to just use input->input->parse_input_event(ev);
```
It's a public method so I thought it might be used by other parts of the code base to push input events directly, but it doesn't seem to be the case: https://github.com/godotengine/godot/search?utf8=%E2%9C%93&q=push_input&type=

What you suggest is what is done for X11, so I guess it could be used for OSX too: https://github.com/godotengine/godot/blob/5b3709d3096df737b8bb2344446be818b0389bfe/platform/x11/os_x11.cpp#L1361

akien-mga commented on platform/osx/os_osx.mm:
```diff
@@ -63,15 +63,12 @@ static NSRect convertRectToBacking(NSRect contentRect) {
-	mod.meta = (flags & NSCommandKeyMask);
-
-	return mod;
+static void get_osx_key_modifier_state(unsigned int flags, Ref<InputEventWithModifiers> state) {
```
Would be better not to hardcode `_osx` in the method name, see e.g. https://github.com/godotengine/godot/blob/5b3709d3096df737b8bb2344446be818b0389bfe/platform/x11/os_x11.cpp#L961
`get_key_modifier_state` would be fine. You could also name the int argument `p_osx_state` to match the X11 reference.

akien-mga commented on platform/osx/os_osx.mm:
```diff
@@ -610,96 +611,101 @@ static int translateKey(unsigned int key) {
-	OS_OSX::singleton->push_input(ev);
-	ev->is_pressed() = false;
-	OS_OSX::singleton->push_input(ev);
+	//toger scroll event has not modifier keys or are they added somewhere else
```
If there are any, I guess you should retrieve them in `(void)scrollWheel:(NSEvent *)event` as done for other events, and pass them as argument to the helper `sendScrollEvent`.

toger5 commented:
Thank you! for the answers!
I used the `input->parse...` function but since OS X is using their builtin event delegate functions I can't access input. that's they there is an additional function...
I should have tried it before asking...

toger5 committed:
added modifier key to scroll event

toger5 commented:
changed the things you mentioned. BUT I also did a pull on upstream master... so there are no multiple commits. is that an issue?

toger5 commented:
@akien-mga I cleaned up the merge I did.
So the pr only has my changes in it now. I decided to make it in two commits because they are different.
c531360 is introducing a new "feature" independent to fixing the InputEvent on osx...

akien-mga commented:
@toger5 I agree about the two commits since implementing modifiers for scroll events is indeed a different topic. On the other hand it would have been better to merge the `get_osx_key_modifier_state` rename in the first commit where it belongs ;)

Also the signature I was proposing was:
```
static void get_key_modifier_state(unsigned int p_osx_state, Ref<InputEventWithModifiers> state)
```

Like for X11:
```
void OS_X11::get_key_modifier_state(unsigned int p_x11_state, Ref<InputEventWithModifiers> state)
```

akien-mga commented:
A good way to know what should go in a commit and what should go in the other is to think: what would have to be reverted if one of the commits were bad? In this case, if the new feature of supporting key modifiers on scroll events had to be reverted, we obviously wouldn't want to revert the unrelated rename of the `get_osx_key_modifier_state` method as well as the removal of your question comment about `push_input`.

toger5 commented:
I'm sorry about the commit structure. I agree it could be better.
I could manually fix it if you want me to?
I at least made the function naming as you want it to...

akien-mga commented:
No it's fine as is, I don't want to be overly annoying ;) Just giving tips for next time.

toger5 commented:
cool,
yea tips are highly appreciated ;)

toger5 commented:
before you merge let me run clang since I now have installed it

toger5 commented:
@akien-mga done.
